AppCenter/AppCenter/Internals/Ingestion/MSOneCollectorIngestion.m

```diff
@@ -94,6 +97,40 @@ - (NSURLRequest *)createRequest:(NSObject *)data
                     stringWithFormat:@"%lld",
                                      (long long)[MSUtility nowInMilliseconds]]
          forKey:kMSOneCollectorUploadTimeKey];
+
+  // Gather tokens from logs.
+  NSMutableString *ticketKeyString = [NSMutableString new];
+  for (id<MSLog> log in container.logs) {
+    MSCommonSchemaLog *csLog = (MSCommonSchemaLog *)log;
+    if (csLog.ext && csLog.ext.protocolExt) {
+      NSArray<NSString *> *ticketKeys = [[[csLog ext] protocolExt] ticketKeys];
+      if (ticketKeys) {
+        for (NSString *ticketKey in ticketKeys) {
+          NSString *token =
+              [[MSTicketCache sharedInstance] ticketFor:ticketKey];
+          if (token) {
+
+            /*
+             * Format to look like this:
+             * "ticketKey1"="d:token1";"ticketKey2"="d:token2".
+             */
+            [ticketKeyString appendString:@"\""];
```

Comment:
This may be the best way in ObjC, but is there not something similar to `String.Format(...)` like in C#?

Reply:
Feels like `stringWithFormat:` would fit better.